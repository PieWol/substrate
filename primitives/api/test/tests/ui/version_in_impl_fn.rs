/// The declaration of the `Runtime` type is done by the `construct_runtime!` macro in a real
/// runtime.
struct Runtime {}

sp_api::decl_runtime_apis! {
	pub trait Api {
		fn test(data: u64);
	}
}

sp_api::impl_runtime_apis! {
	impl self::Api<Block> for Runtime {
		#[api_version(2)]
		fn test(data: u64) {
			unimplemented!()
		}
	}
}

fn main() {}
