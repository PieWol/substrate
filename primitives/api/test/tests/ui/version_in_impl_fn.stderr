error: `api_version` attribute is not allowed `decl_runtime_api` method implementations
  --> tests/ui/version_in_impl_fn.rs:13:3
   |
13 |         #[api_version(2)]
   |         ^
